pkg/server/server.go

```diff
@@ -738,14 +740,27 @@ func (s *ProxyServer) serveRecvBackend(backend Backend, stream agent.AgentServic
 			} else {
 				dialErr := false
 				if resp.Error != "" {
-					klog.ErrorS(errors.New(resp.Error), "DIAL_RSP contains failure", "dialID", resp.Random, "agentID", agentID, "connectionID", resp.ConnectID)
+					// Dial response with error should not contain a valid ConnID.
+					klog.ErrorS(errors.New(resp.Error), "DIAL_RSP contains failure", "dialID", resp.Random, "agentID", agentID)
 					dialErr = true
 				}
 				err := frontend.send(pkt)
 				s.PendingDial.Remove(resp.Random)
 				if err != nil {
 					klog.ErrorS(err, "DIAL_RSP send to frontend stream failure",
 						"dialID", resp.Random, "serverID", s.serverID, "agentID", agentID, "connectionID", resp.ConnectID)
+					// If we never finish setting up the tunnel for ConnectID, then the connection is dead.
+					// Currently, the agent will no resend DIAL_RSP, so connection is dead.
+					// We already attempted to tell the frontend that. We should ensure we tell the backend.
+					pkt := &client.Packet{
+						Type: client.PacketType_CLOSE_REQ,
+						Payload: &client.Packet_CloseRequest{
+							CloseRequest: &client.CloseRequest{
+								ConnectID: resp.ConnectID,
+							},
+						},
+					}
+					_ = stream.Send(pkt)
```

Comment:
I think it's worth logging the error here if it happens?

Reply:
Added a error log a v==5. 
I think its probably worth having for debugging.
However I don't want to exacerbate the alarm.
The earlier "DIAL_RSP send to frontend stream failure" is the more useful and pertinent error.